components/admin_console/system_user_detail/system_user_detail.test.tsx

```diff
@@ -24,7 +25,7 @@ describe('components/admin_console/system_user_detail', () => {
             setNavigationBlocked: jest.fn(),
             addUserToTeam: jest.fn(),
         },
-    };
+    } as any;
```

Comment:
We can use `Props` or `Partial<Props>` here.

Reply:
>We can use Props or Partial<Props> here.

This would require me to mock the whole `RouteComponentProps` interface which I don't think is needed

>Shouldn't need any casting here, if the props are truly optional then they shouldn't be required in the component. Otherwise, they should be added here.

The tests only cover a part of the properties of `User` prop so I don't think we need to add them all, and there's also the problem of `RouteComponentProps` being required, unless I'm missing something

Reply:
What's required from `RouteComponentProps`? If we're not consuming any of the fields in the component then we shouldn't even need to use it.

If we do, we could add those fields to `Props` instead to avoid mocking out the whole `RouteComponentProps` interface.